diff -u generated2/FXApp.pypp.cpp generated/FXApp.pypp.cpp
--- generated2/FXApp.pypp.cpp	2006-06-16 19:25:20.000000000 +0100
+++ generated/FXApp.pypp.cpp	2006-06-16 19:28:50.000000000 +0100
@@ -30,6 +30,23 @@
 using namespace FX;
 namespace bp = boost::python;
 
+static void FXApp_init(FXApp &app, int argc, bp::list argv, unsigned char connect=TRUE)
+{
+    int n, size=PyList_Size(argv.ptr());
+    static QMemArray<const char *> array;
+    array.resize(size+1);
+    for(n=0; n<size; n++)
+    {
+        array[n]=PyString_AsString(PyList_GetItem(argv.ptr(), n));
+    }
+    array[n]=0;
+    app.init(argc, (char **)(array.data()), connect);
+}
+static void FXApp_init2(FXApp &app, int argc, bp::list argv)
+{
+    FXApp_init(app, argc, argv);
+}
+
 struct FXApp_wrapper : FX::FXApp, bp::wrapper< FX::FXApp > {
 
     virtual long int handle( ::FX::FXObject * sender, ::FX::FXSelector sel, void * ptr ) {
